projpicker/web/web.py

```diff
@@ -0,0 +1,170 @@
+        crs_json[f"{crs.crs_auth_name}:{crs.crs_code}"] = entry
+    return crs_json
+
+
+def query(geoms):
+    crs_list = []
+    crs = []
+    if geoms is not None:
+        print(f"{colors.BOLD}ProjPicker query{colors.ENDC}")
+        print(f"{colors.BOLD}-----------------------------{colors.ENDC}")
+        print(geoms)
+        print(f"{colors.BOLD}-----------------------------{colors.ENDC}")
+        parsed_geoms = ppik.parse_mixed_geoms(geoms)
+        crs.extend(ppik.query_mixed_geoms(parsed_geoms))
+        if VERBOSE:
+            ppik.message(f"Query geometries: {parsed_geoms}")
+            ppik.message(f"Number of queried CRSs: {len(crs)}")
+
+    return crs
+
+
+class HttpRequestHandler(http.server.SimpleHTTPRequestHandler):
+    global projpicker_query
+
+    def do_GET(self):
+        if self.path == "/":
+            self.path = "index.html"
+            return http.server.SimpleHTTPRequestHandler.do_GET(self)
+
+        elif self.path == "/projdata":
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+            with open("projdata", "r") as f:
+                self.wfile.write(bytes(f.read(), "utf8"))
+            # print(projpicker_query)
+
+    def do_POST(self):
+        if self.path == "/data":
+            # http.client.HTTPResponse stores headers and implements
+            # email.message.Message class'
+            # https://docs.python.org/3/library/email.compat32-message.html#email.message.Message
+            header = self.headers
+
+            content_len = int(header.get("content-length"))
+            content_charset = header.get_content_charset(failobj="utf-8")
+            content_bytes = self.rfile.read(content_len)
+            content_body = content_bytes.decode(content_charset)
+            projpicker_query = content_body
+
+            geoms = create_parsable_geoms(json.loads(content_body))
+            crs_list = query(geoms)
+            projpicker_query = bbox_to_json(crs_list)
+
+            projpicker_output_json = open("projdata", "w")
+            projpicker_output_json.write(json.dumps(projpicker_query))
+            projpicker_output_json.close()
+
+
+def run(
+    server_class=http.server.HTTPServer,
+    handler_class=HttpRequestHandler,
+    addr="localhost",
+    port=8000,
+    open_in_browser=False,
+):
+    server_address = (addr, port)
+    if addr == "localhost":
+        addr += ":"
+    httpd = server_class(server_address, handler_class)
+
+    try:
+        print(f"{colors.OKGREEN}Starting httpd server on {addr}{port}{colors.ENDC}")
+        if open_in_browser is True:
+            webbrowser.open_new(f"{addr}{port}")
+        httpd.serve_forever()
+    except KeyboardInterrupt:
+        httpd.server_close()
+        print(f"\n{colors.FAIL}Closed httpd server on {addr}:{port}{colors.ENDC}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="ProjPicker webview")
```

Comment:
`ProjPicker HTTP Server` (protocol name and technically more correct?) vs. `ProjPicker Web Server` (more user friendly)